src/dr_llm/storage/_catalog_store.py

```
@@ -270,6 +202,33 @@ def list_models(self, *, query: ModelCatalogQuery) -> list[ModelCatalogEntry]:
                 ).fetchall()
         return [_row_to_entry(row) for row in rows]
 
+    def count_models(self, *, query: ModelCatalogQuery) -> int:
+        self._runtime.init_schema()
+        with self._runtime.conn() as conn:
+            with conn.cursor(row_factory=dict_row) as cur:
```

Ruff (0.15.7)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)